pkg/cache/remote_test.go

```diff
@@ -289,3 +292,37 @@ func TestRemoteCache_MissingBlobs(t *testing.T) {
 		})
 	}
 }
+
+func TestRemoteCache_TLSInsecure(t *testing.T) {
+	type args struct {
+		remoteUrl string
+		insecure  bool
+	}
+
+	tests := []struct {
+		name    string
+		args    args
+		wantErr string
+	}{
+		{
+			name: "happy path - insecure",
+			args: args{
+				remoteUrl: "https://www.example.com",
+				insecure:  true,
+			},
+		},
+		{
+			name: "happy path - secure",
+			args: args{
+				remoteUrl: "https://www.example.com",
+				insecure:  false,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			_, err := cache.NewRemoteCache(cache.RemoteURL(tt.args.remoteUrl), cache.Insecure(tt.args.insecure), http.Header{})
```

Comment:
We should add a test with an insecure HTTPS server.

Reply:
Re-wrote the tests in the latest commits.